nuttx/arch/arm/Kconfig

```diff
@@ -30,176 +30,23 @@
-	select ARCH_HAVE_HEAP2
-	select ARCH_HAVE_LOWVECTORS
-	---help---
-		TI Calypso-based cell phones (ARM7TDMI)
-
-config ARCH_CHIP_DM320
-	bool "TMS320 DM320"
-	select ARCH_ARM926EJS
-	select ARCH_HAVE_LOWVECTORS
-	---help---
-		TI DMS320 DM320 (ARM926EJS)
-
-config ARCH_CHIP_IMX
-	bool "Freescale iMX"
-	select ARCH_ARM920T
-	select ARCH_HAVE_HEAP2
-	select ARCH_HAVE_LOWVECTORS
-	---help---
-		Freescale iMX architectures (ARM920T)
-
-config ARCH_CHIP_KINETIS
-	bool "Freescale Kinetis"
-	select ARCH_CORTEXM4
-	select ARCH_HAVE_MPU
-	select ARCH_HAVE_FPU
-	select ARCH_HAVE_RAMFUNCS
-	---help---
-		Freescale Kinetis Architectures (ARM Cortex-M4)
-
-config ARCH_CHIP_KL
-	bool "Freescale Kinetis L"
-	select ARCH_CORTEXM0
-	select ARCH_HAVE_CMNVECTOR
-	---help---
-		Freescale Kinetis L Architectures (ARM Cortex-M0+)
-
-config ARCH_CHIP_LM
-	bool "TI/Luminary Stellaris"
-	select ARCH_HAVE_CMNVECTOR
-	select ARCH_HAVE_MPU
-	---help---
-		TI/Luminary Stellaris LMS3 and LM4F architectures (ARM Cortex-M3/4)
-
-config ARCH_CHIP_TIVA
-	bool "TI Tiva"
-	select ARCH_HAVE_CMNVECTOR
-	select ARCH_HAVE_MPU
-	select ARCH_HAVE_FPU
-	---help---
-		TI Tiva TM4C architectures (ARM Cortex-M4)
-
-config ARCH_CHIP_LPC17XX
-	bool "NXP LPC17xx"
-	select ARCH_CORTEXM3
-	select ARCH_HAVE_CMNVECTOR
-	select ARCH_HAVE_MPU
-	---help---
-		NXP LPC17xx architectures (ARM Cortex-M3)
-
-config ARCH_CHIP_LPC214X
-	bool "NXP LPC214x"
-	select ARCH_ARM7TDMI
-	select ARCH_HAVE_LOWVECTORS
-	---help---
-		NXP LPC2145x architectures (ARM7TDMI)
-
-config ARCH_CHIP_LPC2378
-	bool "NXP LPC2378"
-	select ARCH_ARM7TDMI
-	select ARCH_HAVE_LOWVECTORS
-	---help---
-		NXP LPC2145x architectures (ARM7TDMI)
-
-config ARCH_CHIP_LPC31XX
-	bool "NXP LPC31XX"
-	select ARCH_ARM926EJS
-	select ARCH_HAVE_LOWVECTORS
-	---help---
-		NPX LPC31XX architectures (ARM926EJS).
-
-config ARCH_CHIP_LPC43XX
-	bool "NXP LPC43XX"
-	select ARCH_CORTEXM4
-	select ARCH_HAVE_CMNVECTOR
-	select ARMV7M_CMNVECTOR
-	select ARCH_HAVE_MPU
-	select ARCH_HAVE_FPU
-	---help---
-		NPX LPC43XX architectures (ARM Cortex-M4).
-
-config ARCH_CHIP_NUC1XX
-	bool "Nuvoton NUC100/120"
-	select ARCH_CORTEXM0
-	select ARCH_HAVE_CMNVECTOR
-	---help---
-		NPX LPC43XX architectures (ARM Cortex-M4).
-
-config ARCH_CHIP_SAMA5
-	bool "Atmel SAMA5"
-	select ARCH_CORTEXA5
-	select ARCH_HAVE_FPU
-	select ARCH_HAVE_LOWVECTORS
-	select ARCH_HAVE_I2CRESET
-	select ARCH_HAVE_TICKLESS
-	select ARCH_HAVE_ADDRENV
-	select ARCH_NEED_ADDRENV_MAPPING
-	---help---
-		Atmel SAMA5 (ARM Cortex-A5)
-
-config ARCH_CHIP_SAMD
-	bool "Atmel SAMD"
-	select ARCH_CORTEXM0
-	select ARCH_HAVE_CMNVECTOR
-	---help---
-		Atmel SAMD (ARM Cortex-M0+)
-
-config ARCH_CHIP_SAM34
-	bool "Atmel SAM3/SAM4"
-	select ARCH_HAVE_MPU
-	select ARCH_HAVE_RAMFUNCS
-	---help---
-		Atmel SAM3 (ARM Cortex-M3) and SAM4 (ARM Cortex-M4) architectures
-
 config ARCH_CHIP_STM32
 	bool "STMicro STM32"
 	select ARCH_HAVE_CMNVECTOR
 	select ARCH_HAVE_MPU
+        select ARCH_HAVE_FPU
```

Comment:
All STM32L4 series chips support FPU and MPU.   I did not want to change the variable config ARCH_CHIP_STM32 because it would require many changes in many external files.   This Kconfig eliminates selectable options for any chips other than those available within the STM32L4 series.